File: backend/Crystal.Core/Entities/Item.cs
namespace Crystal.Core.Entities;

public class Item
{
    public int Id { get; set; }

    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? Distributor { get; set; }
    public string? ImageUrl { get; set; }

    public decimal Price { get; set; }
    public int AlertQuantity { get; set; }
    public DateTime LastUpdate { get; set; }

    public Book? Book { get; set; }
    public bool IsActive { get; set; }
    public ICollection<InventoryLine> InventoryLines { get; set; } = new List<InventoryLine>();
    public ICollection<Receipt> Receipts { get; set; } = new List<Receipt>();
}

File: frontend/src/data/types/inventory/item.ts
interface BaseItemFields {
	id: number;
	name: string;
	description: string | null;
	imageUrl: string | null;
	price: number;
	totalQuantity: number;
	alertQuantity: number;
	isLowStock: boolean;
	lastUpdate: string;
	isBook: boolean;
	isActive: boolean;
}

export interface ItemApiDTO extends BaseItemFields {
	distributor?: string | null;
	isbn?: string | null;
	publicationDate?: string | null;
	authors?: string[];
	authorIds?: number[];
	publishers?: string[];
	categories?: string[];
	categoryIds?: number[];
}

export interface Item extends BaseItemFields {
	distributor: string | null;
	isbn: string | null;
	publicationDate: string | null;
	authors: string[];
	authorIds?: number[];
	publishers: string[];
	categories: string[];
	categoryIds: number[];
}

export interface CreateItemRequest {
	name: string;
	description: string | null;
	distributor?: string | null;
	price: number;
	alertQuantity: number;
	imageUrl?: null;
}

export interface CreateBookRequest extends CreateItemRequest {
	isbn: string;
	publicationDate: string;
	authors?: string[];
	publishers?: string[];
	authorIds: number[];
	categoryIds: number[];
	publisherIds: number[];
}
